project-overview.md

```diff
@@ -0,0 +1,69 @@
+
+ndquirk is a proposal for an extensible tool that quickly identifies quirks in N-dimensional datasets, where a 'quirk' is a deviation from a defined set of expectations.
+
+## Goals
+Primary goal
+- Provide a accessible, extensible, and performant tool for identifying and communicating data deviations from a set of expectations for N-D datasets.
+
+Sub-goals
+- Provide simple mechanisms to run validation proximal to the data.
+- Provide mechanisms for distributed data validation.
+- Leverage existing software and validators whenever possible.
+- Minimize as much as possible the number of get requests and amount of data transferred during data validation.
+- Provide simple storage option configuration (i.e., no pass-through kwargs).
+- Provide simple run-time configuration.
+- Provide accessible documentation and demonstrations for all common use cases.
+- Maintain 100% coverage in testing utilities.
+
+## Partner and Stakeholders
+
+This utility would be relevant for any data provider or user of N-D data. For example, the goals should include making the tool accessible and useful for data providers at NASA, NOAA, and USGS.
+
+This proposal has grown from many conversations over the years, all of whom would be likely either partners or stakeholders in the project. These people include:
+
+- @abarciauskas-bgse
+- @sharkinsspatial
+- @jhamman
+- @norlandrhagen
+- @rabernat
+- @briannapagan
+- @andersy005
+- @eni-awowale
+- @omshinde
+
+## Milestones
+
+- [ ] Define project goals
+    - [ ] Construct a list of use cases for the library
+    - [ ] Define stakeholders and communication pathways for soliciting feedback
+- [ ] Complete an API design document, including establishing whether a new library is necessary or an extension for great expectations would be sufficient
+- [ ] Implement a mechanism for defining and connecting to a data source
+- [ ] Implement, test, and document mechanisms for defining expectations
+	- [ ] Demonstrate defining data format expectation (e.g., Cloud-Optimized GeoTIFF)
+		- [ ] Define mechanisms for versioning data and metadata formats expectations
+	- [ ] Demonstrate defining a chunk schema expectation (e.g., constant chunk sizes)
+	- [ ] Demonstrate defining an encoding expectation (e.g., entirely zlib compression)
+	- [ ] Demonstrate defining metadata standards (e.g., CF, Croissant)
+	- [ ] Demonstrate defining data hierarchy expectations (e.g., Dataset vs. DataTree)
+	- [ ] Demonstrate defining data type expectations (e.g., all float64)
+	- [ ] Demonstrate defining data value expectations (e.g., no NaNs)
+- [ ] Demonstrate validating expectations in parallel
+- [ ] Demonstrate running the expectations using data proximate computing
+- [ ] Demonstrate building a browser-based UI for defining and validating expectations
+
+### Resources
+
+- [https://github.com/great-expectations/great_expectations/issues/1942](https://github.com/great-expectations/great_expectations/issues/1942)
```

Comment:
I think if you took xarray-schema and added support for types of validation that xarray could not perform (e.g. union types, functional validators) then you would end up with something that's pretty similar to calling great expectations on a pandas dataframe

https://docs.greatexpectations.io/docs/0.18/oss/guides/connecting_to_your_data/fluent/in_memory/connect_in_memory_data/

Reply:
Oh or like Pandera I guess

Reply:
This approach would work off-the-bat for a subset of the expectations, but I don't expect it to be entirely sufficient. xarray-schema currently relies on having loaded or at least opened an xarray dataset for validation. IMO a lot of the value of this tool would be providing explanations for why datasets cannot be simply opened with Xarray (e.g., https://github.com/briannapagan/quirky-data-checker/blob/main/results/results_GES_DISC_total_quirks.png)